movieApp.js

```diff
@@ -1,170 +1,264 @@
+            movieDetails.style.display = "none";
+          } else {
+            movieDetails.style.display = "block";
+          }
+        }
+      }
 
-  readMoreBtn.addEventListener("click", function onclick() {
-    this.classList.toggle("active");
-    /* Toggle between hiding and showing the active panel */
-    if (movieDetails.style.display === "block") {
-      movieDetails.style.display = "none";
-    } else {
-      movieDetails.style.display = "block";
+      readMoreBtn.addEventListener("click", toggle);
+      return movieContainer;
     }
-  });
-  return movieContainer;
-}
 
-function createMoviesPage() {
-  nowStreaming.innerHTML = "";
-  movies.forEach((movie) => {
-    const movieElement = createMovieCard(movie);
-    nowStreaming.appendChild(movieElement);
-  });
+    function createMoviesPage() {
+      nowStreaming.innerHTML = "";
+      movies.forEach((movie) => {
+        const movieElement = createMovieCard(movie);
+        nowStreaming.appendChild(movieElement);
+      });
+    }
+    createMoviesPage();
+    // JS 2
+    const searchBtn = document.getElementById("search-icon");
+
+    // function to search
+    const inputField = document.getElementById("keyword-search");
+    searchBtn.addEventListener("click", () => {
+      const inputStr = inputField.value.toLowerCase();
+      nowStreaming.innerHTML = "";
+      let found = false; // Flag to check if any movie is found
+
+      movies.forEach((movie) => {
+        if (
+          movie.title.toLowerCase().includes(inputStr) ||
+          movie.description.toLowerCase().includes(inputStr)
+        ) {
+          found = true;
+          const searchMovieElement = createMovieCard(movie);
+          nowStreaming.appendChild(searchMovieElement);
+        }
+        if (!found) {
+          nowStreaming.innerHTML = `<p>Not Found</p>`;
+        }
+      });
+    });
+
+    //
+    // creating toggling in filter/shorting option
+    const shorting = document.getElementById("shorting");
+    const filters = document.getElementById("filters");
+    shorting.addEventListener("mouseover", function () {
+      this.classList.toggle("active");
+      if (filters.style.display === "block") {
+        filters.style.display = "none";
+      } else {
+        filters.style.display = "block";
+      }
+    });
+    const newRelease = document.getElementById("newRelease");
+    newRelease.addEventListener("click", function () {
+      nowStreaming.innerHTML = "";
+      movies.forEach((movie) => {
+        if (movie.movie_year >= 2023) {
+          const movieElement = createMovieCard(movie);
+          nowStreaming.appendChild(movieElement);
+        }
+      });
+    });
+
+    const login = document.getElementById("loginButton");
+    const loginPage = document.getElementById("logIn");
+
+    login.addEventListener("mouseover", function onclick() {
+      this.classList.toggle("active");
+      /* Toggle between hiding and showing the active panel */
```

Comment:
Try to be consistent with your comments style. Also, it's always better to leave an empty line above a line of comment, to make the code more readable.

Reply:
noted